REB_firmware/REB_firmware_update/v0/producer_REB_firmware_update.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+from ccsTools import ccsProducer
+from java.lang import Exception
+import Tkinter
+import glob
+import os
+import subprocess
+
+CCS.setThrowExceptions(True);
+
+#Copy firmware_load_scripts files
+instdir = os.getenv('LCATR_INSTALL_AREA').replace('share/','')
+os.system('cp -vp %s/REB_v5/firmware_load_scripts/bin/et_prog_flash.sh .', instdir)
+os.system('cp -vp %s/REB_v5/firmware_load_scripts/bin/flash_load.tcl .', instdir)
+
+#Safety button
+apptxt = "Click here when ready to start firmware update of the REB"
+print apptxt
+top = Tkinter.Tk()
+A = Tkinter.Button(top, text = apptxt, command = top.destroy, bg = "yellow", font = ("Helvetica",24))
+A.pack()
+top.title('Are you ready to start the firmware update?')
+top.mainloop()
+
+#Get list of files in firmware directory
+list_of_firmware_files = glob.glob(os.path.join(instdir,'/REB_v5/targets/REB_v5_top/images/', '*.mcs'))
+
+#Find most recent file
+recent_file = max(list_of_firmware_files, key=os.path.getctime)
```

Comment:
How we imagine this working in practice: 
1) Technical expert tells the operator that they would like to have the firmware upgraded.
2) Technical expert connects all equipment and powers on the REBs.
3) Operator proceeds through the eTraveler.
4) Technical expert powers down the REBs and disconnects the equipment.

Our operators are, in many cases, non-technical people. They aren't asked to make any comparisons or decisions, just proceed through eTravelers. The decision to upgrade the REB will be made before the operator even arrives.

If you are asking why I have an "upgrade selector" step in the eTraveler, it is in case a technical expert wants to use the eTraveler as a way to find out what firmware version is currently installed on the REB, but doesn't necessarily want to upgrade. Operators would only be using the eTraveler if they are instructed by a technical expert to perform and upgrade, so of course they will always be selecting "upgrade necessary."